packages/flutter_tools/lib/src/usage.dart

```diff
@@ -25,10 +24,14 @@ class Usage {
     final String version = versionOverride ?? FlutterVersion.getVersionString(whitelistBranchName: true);
     _analytics = new AnalyticsIO(_kFlutterUA, settingsName, version);
 
-    // Report a more detailed OS version string than package:usage does by
-    // default. Also, send the branch name as the "channel".
+    // Report a more detailed OS version string than package:usage does by default.
     _analytics.setSessionValue('cd1', os.name);
+    // Send the branch name as the "channel".
     _analytics.setSessionValue('cd2', FlutterVersion.getBranchName(whitelistBranchName: true));
+    // Record the host as the application installer ID - the context that flutter_tools is running in.
+    if (platform.environment.containsKey('FLUTTER_HOST')) {
```

Comment:
There's more context here: https://github.com/flutter/flutter/issues/8989, but IDEs set it when invoking `flutter` (specifically so the flutter tools can use it to categorize its analytics).